doc/reference/kernel/threads/index.rst

```diff
@@ -341,7 +341,11 @@ The following code has the same effect as the code segment above.
 
     K_THREAD_DEFINE(my_tid, MY_STACK_SIZE,
                     my_entry_point, NULL, NULL, NULL,
-                    MY_PRIORITY, 0, K_NO_WAIT);
+                    MY_PRIORITY, 0, 0);
+
+.. note::
+   When declaring a thread at compile time it is not possible to use the
+   ``K_NO_WAIT`` macro for the ``delay`` parameter. Use 0 instead.
```

Comment:
I stumbled upon this issue, but with `K_FOREVER`. Maybe add it to the note?

Reply:
I think `K_FOREVER` is just not a logical value to pass to a delay in starting a thread?

Reply:
```
	 * Note that static threads defined using the legacy API have a
	 * delay of K_FOREVER.
	 */
	k_sched_lock();
	_FOREACH_STATIC_THREAD(thread_data) {
		if (thread_data->init_delay != K_TICKS_FOREVER) {
			schedule_new_thread(thread_data->init_thread,
```
It appears you can use `K_TICKS_FOREVER` to not schedule a thread at all.
There must be a manual way to schedule the thread then.

Reply:
yes, you can start it with `k_thread_start()`. We have some test code that does this.